plugins/outputs/sql/sql.go

```diff
@@ -98,7 +98,13 @@ func (p *SQL) Init() error {
 }
 
 func (p *SQL) Connect() error {
-	db, err := gosql.Open(p.Driver, p.DataSourceName)
+	dsnBuffer, err := p.DataSourceName.Get()
+	if err != nil {
+		return fmt.Errorf("loading data source name secret failed: %w", err)
+	}
+
+	db, err := gosql.Open(p.Driver, dsnBuffer.String())
+	dsnBuffer.Destroy()
```

Comment:
```suggestion
	dsnBuffer, err := p.DataSourceName.Get()
	if err != nil {
		return fmt.Errorf("loading data source name secret failed: %w", err)
	}
	dsn := dsnBuffer.String()
	dsnBuffer.Destroy()

	db, err := gosql.Open(p.Driver, dsn)
```

Reply:
Yes and it was correct until you requested it to be changed! With your change, the locked memory pages are held way longer without _any_ benefit!

Reply:
Benefit is the DSN doesn't stay in memory in plaintext during the whole execution of this method :wink: